src/core/lib/iomgr/executor.h

```diff
@@ -21,30 +21,66 @@
 
 #include <grpc/support/port_platform.h>
 
+#include "src/core/lib/gpr/spinlock.h"
+#include "src/core/lib/gprpp/thd.h"
 #include "src/core/lib/iomgr/closure.h"
 
+typedef struct {
```

Comment:
You don't need `typedef` in C++. Simply do `struct thread_state {...}`.

In addition, I'd change the type name to comply with [the style guide](https://google.github.io/styleguide/cppguide.html#Aliases): `ThreadState`.

Reply:
Done.